Dockerfile

```diff
@@ -12,21 +12,42 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# About: This dockerfile builds the cip binary for use in auditor e2e tests.
+# About: This dockerfile builds the cip binary for auditor tests and production use.
+#
+# Usage: Since there are two variants to build, you must include the variant name during build-time.
+# cip production binary: 
+#   docker build --build-arg variant=cip /path/to/Dockerfile
+# test auditor:
+#   docker build --build-arg variant=auditor /path/to/Dockerfile
+
+# Determine final build variant [cip | auditor].
+ARG variant
 
-FROM golang:latest
+FROM golang:latest AS base
 # Transfer all project files to container.
 WORKDIR /go/src/app
 COPY . .
 # Build and export cip command.
 RUN ./go_with_version.sh build ./cmd/cip
 RUN cp ./cip /bin/cip
-# Include cip-auditor testing fixtures.
-RUN mkdir /e2e-fixtures
-RUN cp -r ./test-e2e/cip-auditor/fixture/* /e2e-fixtures
 # Provide docker config for repo information.
 RUN mkdir /.docker
 RUN cp ./docker/config.json /.docker/config.json
+
+FROM google/cloud-sdk:latest AS gcloud-base
```

Comment:
so there was an effort of removing most dependencies on DockerHub last year (ref: https://github.com/kubernetes/sig-release/issues/1369) and I think this might fall into the same category.

is there a non-DockerHub (maybe GCR?) counterpart for `cloud-sdk` image? if so, I would love it if we can use that 😄 ditto on the `golang` image but that one is not net-new so I'm more okay if we defer that one until we have stronger reasons to do it 😄

Reply:
Absolutely! Thank you for bringing this up. I can use this image instead:
`gcr.io/google.com/cloudsdktool/cloud-sdk:latest`